source/common/upstream/cluster_manager_impl.cc

```diff
@@ -924,7 +924,7 @@ void ClusterManagerImpl::updateClusterCounts() {
   if (all_clusters_initialized && ads_mux_) {
     const auto type_url = Config::getTypeUrl<envoy::config::cluster::v3::Cluster>();
     const uint64_t previous_warming = cm_stats_.warming_clusters_.value();
-    if (previous_warming == 0 && !warming_clusters_.empty()) {
+    if (!warming_clusters_.empty()) {
```

Comment:
`previous_warming` cant be 0 when `updateClusterCounts()` in invoked from `onClusterInit()` i.e after warming is done. This is because `loadCluster()` is also invoking `updateClusterCounts()` which increases it before the warming starts.
```
 const auto previous_cluster =
      loadCluster(cluster, new_hash, version_info, true, warming_clusters_);
  auto& cluster_entry = warming_clusters_.at(cluster_name);
  if (!all_clusters_initialized) {
    ENVOY_LOG(debug, "add/update cluster {} during init", cluster_name);
    init_helper_.addCluster(*cluster_entry);
  } else {
    ENVOY_LOG(debug, "add/update cluster {} starting warming", cluster_name);
    cluster_entry->cluster_->initialize([this, cluster_name] {
      ENVOY_LOG(debug, "warming cluster {} complete", cluster_name);
      auto state_changed_cluster_entry = warming_clusters_.find(cluster_name);
      onClusterInit(*state_changed_cluster_entry->second);
```

Reply:
My bad. Pausing cds ack is working already. 
>previous_warming cant be 0 when updateClusterCounts() in invoked from onClusterInit() i.e after warming is done. This is because loadCluster() is also invoking updateClusterCounts() which increases it before the warming starts.

This is true but it does not stop/break pausing of cds ack. cds ack gets paused from the `loadCluster()` itself.